blog/2023/02/the-injectivity-radii-of-the-unitary-groups.md

```diff
@@ -0,0 +1,206 @@
+Since these groups are featured in many (all?) of the introductory texts on Lie groups and manifolds, and since the injectivity radius is a basic property introduced in differential geometry textbooks, I was surprised that I could not find a single reference giving these radii for these groups.
+
+In this post I'll work out these radii.
+Marvelously, we don't need any differential geometry or group theory to do this, just linear algebra!
+Nevertheless, this post assumes familiarity with these topics and for the sake of space will try not define all common terms or notation.[^2]
+
+## The injectivity radius
+
+Consider a point $p$ on some manifold $\mathcal{M}$ with tangent vectors $X,Y \in T_p \mathcal{M}$.
+Assume a Riemannian metric $g_p$ defining an inner product $g_p\colon (X, Y) \mapsto \ip{X}{Y}_g$, which induces a norm $\norm{X}_g$.
+
+Let's denote the exponential map $\exp_p\colon X \mapsto q$ for $q \in \mathcal{M}$ and the logarithmic map $\log_p\colon q \mapsto Y$.
+The injectivity radius at $p$ is defined as the norm of the smallest $X$ for which $X \ne Y = \log_p (\exp_p X)$, or in other words, the smallest $X$ for which the logarithmic map no longer is the inverse of the exponential map.
+We further define the global injectivity radius of $\mathcal{M}$ as the infimum of the injectivity radii at all points on the manifold.
+
+Notationally, we define this global injectivity radius as
+$$\operatorname{inj}^-_{\mathcal{M}} = \inf_{(p, X) \in T \mathcal{M}}\{\norm{X}_g | \log_p(\exp_p(X)) \ne X\}.$$
+
+We'll also consider the related supremum
+$$\operatorname{inj}^+_{\mathcal{M}} = \sup_{(p, X) \in T \mathcal{M}}\{\norm{X}_g | \log_p(\exp_p(X)) = X\}.$$
+These two quantities form the lower and upper bound radii, respectively, of two geodesic balls within which the exponential map is invertible.
+
+## The unitary group(s)
+
+The Unitary group $\mathrm{U}(n, \mathbb{F})$ over some number system $\mathbb{F}$ is the group of all $n \times n$ matrices $p \in \mathbb{F}^{n \times n}$ for which $p^\mathrm{H} p = I_n$, where ${\cdot}^\mathrm{H}$ denotes the matrix adjoint.
+$\mathbb{F}$ could be the real numbers $\R$, complex numbers $\C$, or quaternions $\H$.
+
+We will also deal with the following subgroups:
+- $\mathrm{SU}(n)$: the special unitary group, which consists of complex unitary matrices whose determinant is +1
+- $\mathrm{O}(n) \equiv \mathrm{U}(n, \R)$: the orthogonal group, i.e. the group of rotations and reflections
+- $\mathrm{SO}(n)$: the special orthogonal group, i.e. the group of real rotations
+
+We will focus on the real and complex fields, but the unitary quaternionic case immediately follows from the complex one.
+
+## Relevant geometric properties
+
+The unitary group is a compact group and when equipped with the Euclidean inner product $g\colon (X, Y) \mapsto \ip{X}{Y}$ becomes a Riemannian manifold.
```

Comment:
Since X and Y are from the Lie group and skew-symmetric (skew -Hermitian) many people use a factor 1/2 upfront (Hüper for example) but we did not in Manifolds.jl because we use the same embedding idea as you do here). Should we mention that here?

Reply:
Similarly a note why it is enough to look at the Lie algebra might be worth a short note somewhere?

Reply:
At this point X and Y are elements of any tangent space and not restricted to the Lie algebra. That's introduced a few lines below.

But agreed, I should add a note about the 1/2 scale and probably also say "Frobenius inner product"

Reply:
Ah that's why, because it is introduced below. I am more like a linear reader, that is things should be introduced before they are used.

Reply:
No, the tangent space was introduce above when we defined the injectivity radius. We haven't mentioned the Lie algebra yet.

Reply:
Ah but I also meant to really _say_ what the tangent space here looks like, unless it's enough to really state that for the Lie algebra. That might make this story a little more concrete already at this point.

Reply:
Sure that sounds reasonable, I am just thinking what might be a good reason to avoid that the reader thinks about this when you introduce injectivity and tangent spaces (while its easier later). Or maybe the reader has to be patient until at the Lie algebra he/she notices: Ah looking at that is enough.